pkg/controller/cyclenoderequest/transitioner/transitions.go

```diff
@@ -106,6 +109,23 @@ func (t *CycleNodeRequestTransitioner) transitionPending() (reconcile.Result, er
 
 	// get instances inside cloud provider node groups
 	nodeGroupInstances := nodeGroups.Instances()
+
+	forceCheckAsgConfigAnnotations := t.cycleNodeRequest.Annotations[ForceAsgConfigCheckAnnotation]
+	if forceCheckAsgConfigAnnotations == "true" {
+		var skipped = true
+		for _, instance := range nodeGroupInstances {
+			outOfDate := instance.OutOfDate()
+			if outOfDate {
+				skipped = false
```

Comment:
what you could do instead of a bool is just have the return directly in the outOfDate case;

```go
if outOfDate {
  t.cycleNodeRequest.Status.Message = "Cycling skipped because all nodes in the nodegroups are up-to-date with a cloud provider ASG"
  return t.transitionObject(v1.CycleNodeRequestSkipped)
}
```

Will keep the rest of the function after clean